### ISSUE TITLE
New rule suggestion: no-restricted-attributes

### ISSUE THREAD

**Issue by JulianMaus98** [angular-eslint/angular-eslint#1265]
Hi, while integrating the NgOptimizedImage directive (https://angular.io/guide/image-directive) I searched for a way to restrict the usage of the `src` attribute in the template in favour of the `ngSrc` attribute.
This can be done via the rule "no-restricted-attrs" of another project: "HTML ESlint" (https://yeonjuan.github.io/html-eslint/docs/rules/no-restricted-attrs/), hovever, when already using "Angular ESLint" to lint HTML files, it is not possible to combine the parsers of both packages. So I think it is useful - not only for this use case - to provide such functionality in "Angular ESlint", too.

The provided documentation of the other project (https://yeonjuan.github.io/html-eslint/docs/rules/no-restricted-attrs/) contains an example of how to use the rule (I'll provide another example of how it could look like below) and I think it is a very convenient way to configure which tags are allowed and which are not.

```JSON
 {
  "rules": {
    "@html-angular/no-restricted-attributes": ["warn",  {
      "tagPatterns": ["^img$"],
      "attrPatterns": ["src.*"]
      "message": "consider using ngSrc instead of src"
  }]
  }
}

```

```HTML
// Warning
<img src="imageSource">

// Correct
<img ngSrc="imageSource">
```

**Versions**

| package                                   | version |
| ----------------------------------        | ------- |
| `@angular-eslint/eslint-plugin-template`  | `15.0.0` |
| `@angular-eslint/template-parser`         | `15.0.0` |
| `@typescript-eslint/parser`               | `5.43.0` |
| `ESLint`                                  | `8.27.0` |
| `node`                                    | `16.17.0` |

```sh
# Please run `npx ng version` in your project and paste the full output here:
Angular CLI: 15.0.0
Node: 16.17.0
Package Manager: npm 8.19.2
OS: linux x64

Angular: 
... 

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.1500.0 (cli-only)
@angular-devkit/core         15.0.0 (cli-only)
@angular-devkit/schematics   15.0.0 (cli-only)
@schematics/angular          15.0.0 (cli-only)
```

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by strawberry-choco** [comment]
Any updates on this? I can provide write implementation if needed.

**Comment by jerone** [comment]
This would be useful to have, but maybe as a dedicated rule "use-optimized-image", instead of a generic rule "no-restricted-attributes". Advantage would be that one can simply enable the rule, instead of having to configure it as seen in the first post.

**Comment by strawberry-choco** [comment]
Alright. I will provide an implementation as its own rule soonish. 🙂

**Comment by JulianMaus98** [comment]
> Alright. I will provide an implementation as its own rule soonish. 🙂

Thanks a lot!
